### PR TITLE
feat(top-bar): create a top-bar component

### PR DESCRIPTION
**Pull request by phiz71**
**Issue**

https://graviteecommunity.atlassian.net/browse/APIM-301

**Description**

Creation of a top bar component

---
📚&nbsp;&nbsp;View the storybook of this branch [here](https://6183b02d73381a003a3be1a6-dcgboagjln.chromatic.com)

### CODE REVIEW

**Comment by graviteeio** [comment]
**Gravitee.io Automatic Deployment**
A prerelease version of this package has been published! 🚀
To use it in your project run the following lines:
```shell
npm install @gravitee/ui-policy-studio-angular@4.5.2-gio-top-bar-48405cd
# or
yarn add @gravitee/ui-policy-studio-angular@4.5.2-gio-top-bar-48405cd
```

**Comment by graviteeio** [comment]
**Gravitee.io Automatic Deployment**
A prerelease version of this package has been published! 🚀
To use it in your project run the following lines:
```shell
npm install @gravitee/ui-particles-angular@4.5.2-gio-top-bar-48405cd
# or
yarn add @gravitee/ui-particles-angular@4.5.2-gio-top-bar-48405cd
```

**Comment by graviteeio** [comment]
**Gravitee.io Automatic Deployment**
A prerelease version of this package has been published! 🚀
To use it in your project run the following lines:
```shell
npm install @gravitee/ui-particles-angular@4.5.3-gio-top-bar-cbdc5ab
# or
yarn add @gravitee/ui-particles-angular@4.5.3-gio-top-bar-cbdc5ab
```

**Comment by graviteeio** [comment]
**Gravitee.io Automatic Deployment**
A prerelease version of this package has been published! 🚀
To use it in your project run the following lines:
```shell
npm install @gravitee/ui-particles-angular@4.5.3-gio-top-bar-6b252d8
# or
yarn add @gravitee/ui-particles-angular@4.5.3-gio-top-bar-6b252d8
```

**Comment by graviteeio** [comment]
**Gravitee.io Automatic Deployment**
A prerelease version of this package has been published! 🚀
To use it in your project run the following lines:
```shell
npm install @gravitee/ui-policy-studio-angular@4.5.3-gio-top-bar-6b252d8
# or
yarn add @gravitee/ui-policy-studio-angular@4.5.3-gio-top-bar-6b252d8
```

**Comment by graviteeio** [comment]
**Gravitee.io Automatic Deployment**
A prerelease version of this package has been published! 🚀
To use it in your project run the following lines:
```shell
npm install @gravitee/ui-policy-studio-angular@4.5.3-gio-top-bar-4758179
# or
yarn add @gravitee/ui-policy-studio-angular@4.5.3-gio-top-bar-4758179
```

**Comment by graviteeio** [comment]
**Gravitee.io Automatic Deployment**
A prerelease version of this package has been published! 🚀
To use it in your project run the following lines:
```shell
npm install @gravitee/ui-particles-angular@4.5.3-gio-top-bar-4758179
# or
yarn add @gravitee/ui-particles-angular@4.5.3-gio-top-bar-4758179
```

**Comment by graviteeio** [comment]
**Gravitee.io Automatic Deployment**
A prerelease version of this package has been published! 🚀
To use it in your project run the following lines:
```shell
npm install @gravitee/ui-particles-angular@4.7.0-gio-top-bar-89ffea3
# or
yarn add @gravitee/ui-particles-angular@4.7.0-gio-top-bar-89ffea3
```

**Comment by graviteeio** [comment]
**Gravitee.io Automatic Deployment**
A prerelease version of this package has been published! 🚀
To use it in your project run the following lines:
```shell
npm install @gravitee/ui-policy-studio-angular@4.7.0-gio-top-bar-89ffea3
# or
yarn add @gravitee/ui-policy-studio-angular@4.7.0-gio-top-bar-89ffea3
```

**Comment by graviteeio** [comment]
**Gravitee.io Automatic Deployment**
A prerelease version of this package has been published! 🚀
To use it in your project run the following lines:
```shell
npm install @gravitee/ui-policy-studio-angular@4.7.0-gio-top-bar-f016a8e
# or
yarn add @gravitee/ui-policy-studio-angular@4.7.0-gio-top-bar-f016a8e
```

**Review by gcusnieux — APPROVED** [review]

File: ui-particles-angular/projects/ui-particles-angular/src/lib/gio-top-bar/gio-top-bar.component.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+import { GioTopBarProductTypes } from './gio-top-bar-product/gio-top-bar-product.component';
+
+@Component({
+  selector: 'gio-top-bar',
+  templateUrl: './gio-top-bar.component.html',
+  styleUrls: ['./gio-top-bar.component.scss'],
+})
+export class GioTopBarComponent {
+  @Input() public type: GioTopBarProductTypes = 'am';
```
Comment (ThibaudAV) [inline ui-particles-angular/projects/ui-particles-angular/src/lib/gio-top-bar/gio-top-bar.component.ts:26]: I don't really understand why we have the `type` and the component extends 🤔 

I will propose either one or the other.
If we have one type then we don't need to make one component per product. 
And if we want a component per product then no type we use the product component directly in the html (which would avoid the product component loading the code of another product)

File: ui-particles-angular/projects/ui-particles-angular/src/lib/gio-top-bar/gio-top-bar.component.scss [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    width: 36px;
+    height: 36px;
+    padding: 6px;
+    border-radius: 4px;
+    cursor: pointer;
+  }
+}
+
+.gio-top-bar__icon:hover {
+  background: map.get(gio.$mat-space-palette, lighter20);
```
Comment (ThibaudAV) [inline ui-particles-angular/projects/ui-particles-angular/src/lib/gio-top-bar/gio-top-bar.component.scss:43]: How about adding properly a button ? 
```
<button mat-icon-button class="example-icon" aria-label="Example icon-button with menu icon">
    <mat-icon>menu</mat-icon>
  </button>
```
Play with cursor to simulate a button is not a good practice I think 😁

File: ui-particles-angular/projects/ui-particles-angular/src/lib/gio-top-bar/gio-top-bar.component.scss [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+:host {
+  width: 100%;
+}
+
+.gio-top-bar {
+  display: flex;
+  align-items: center;
+  padding: 16px;
+  background: map.get(gio.$mat-space-palette, lighter10);
+  color: map.get(gio.$mat-dove-palette, default);
```
Comment (ThibaudAV) [inline ui-particles-angular/projects/ui-particles-angular/src/lib/gio-top-bar/gio-top-bar.component.scss:30]: Why  the colour is not the contrast colour of the background 🤔 ?

**Comment by graviteeio** [comment]
**Gravitee.io Automatic Deployment**
A prerelease version of this package has been published! 🚀
To use it in your project run the following lines:
```shell
npm install @gravitee/ui-policy-studio-angular@4.7.0-gio-top-bar-6d72ffd
# or
yarn add @gravitee/ui-policy-studio-angular@4.7.0-gio-top-bar-6d72ffd
```

**Comment by graviteeio** [comment]
**Gravitee.io Automatic Deployment**
A prerelease version of this package has been published! 🚀
To use it in your project run the following lines:
```shell
npm install @gravitee/ui-particles-angular@4.7.0-gio-top-bar-6d72ffd
# or
yarn add @gravitee/ui-particles-angular@4.7.0-gio-top-bar-6d72ffd
```

File: ui-particles-angular/projects/ui-particles-angular/src/lib/gio-top-bar/gio-top-bar.component.scss (hunk 3 above)
Comment (ThibaudAV) [reply ui-particles-angular/projects/ui-particles-angular/src/lib/gio-top-bar/gio-top-bar.component.scss:47]: maybe we can ping @audrey-tlm here 😁

**Review by ThibaudAV — APPROVED** [review]

**Comment by graviteeio** [comment]
**Gravitee.io Automatic Deployment**
A prerelease version of this package has been published! 🚀
To use it in your project run the following lines:
```shell
npm install @gravitee/ui-policy-studio-angular@4.7.0-gio-top-bar-b1f6ba1
# or
yarn add @gravitee/ui-policy-studio-angular@4.7.0-gio-top-bar-b1f6ba1
```

**Comment by graviteeio** [comment]
**Gravitee.io Automatic Deployment**
A prerelease version of this package has been published! 🚀
To use it in your project run the following lines:
```shell
npm install @gravitee/ui-particles-angular@4.7.0-gio-top-bar-b1f6ba1
# or
yarn add @gravitee/ui-particles-angular@4.7.0-gio-top-bar-b1f6ba1
```

**Comment by graviteeio** [comment]
**Gravitee.io Automatic Deployment**
A prerelease version of this package has been published! 🚀
To use it in your project run the following lines:
```shell
npm install @gravitee/ui-particles-angular@4.7.0-gio-top-bar-4dad30d
# or
yarn add @gravitee/ui-particles-angular@4.7.0-gio-top-bar-4dad30d
```

**Comment by graviteeio** [comment]
**Gravitee.io Automatic Deployment**
A prerelease version of this package has been published! 🚀
To use it in your project run the following lines:
```shell
npm install @gravitee/ui-policy-studio-angular@4.7.0-gio-top-bar-4dad30d
# or
yarn add @gravitee/ui-policy-studio-angular@4.7.0-gio-top-bar-4dad30d
```

**Comment by graviteeio** [comment]
**Gravitee.io Automatic Deployment**
A prerelease version of this package has been published! 🚀
To use it in your project run the following lines:
```shell
npm install @gravitee/ui-policy-studio-angular@4.7.0-gio-top-bar-8d6f90f
# or
yarn add @gravitee/ui-policy-studio-angular@4.7.0-gio-top-bar-8d6f90f
```

**Comment by graviteeio** [comment]
**Gravitee.io Automatic Deployment**
A prerelease version of this package has been published! 🚀
To use it in your project run the following lines:
```shell
npm install @gravitee/ui-particles-angular@4.7.0-gio-top-bar-8d6f90f
# or
yarn add @gravitee/ui-particles-angular@4.7.0-gio-top-bar-8d6f90f
```

**Pull request merged**